### DISCUSSION TITLE
How to animate a popover with framer motion?

### DISCUSSION THREAD

**Discussion by DoneDeal0** [ariakit/ariakit#843] [Questions] [answered]
Hello,

I'm trying to animate a popover with framer motion. I would like to make it appear smoothly with opacity, and if possible with a scale or clip path effect. But just opacity would be great for a start. 

I've read the chat on spectrum as well as your exemple based on react-spring on your official doc. I can't make it work. Here is a sandbox that reproduce a minimum setup: https://codesandbox.io/s/kind-lederberg-xkyn2?file=/src/App.tsx:0-811

How would you do it? Thanks!

**Comment by diegohaz** [comment, marked as answer]
I don't have much experience with Framer Motion, but here's a working example I managed to build after briefly reading the docs: https://codesandbox.io/s/reakit-with-framer-motion-z86rb?file=/src/App.tsx

**Reply by DoneDeal0** [reply]
Hi Diego! Sorry for my late reply, I wasn't notified of your response, I only see the result now. This is a great answer, thank you!

**Comment by joseDaKing** [comment]
I am maybe late to the party but here is a clean solution, you could use the same logic for react-spring. I hope this would help other people.

```tsx
import React from "react";

import { PopoverDisclosure, Popover, usePopoverState } from "reakit";

import { AnimatePresence, motion } from "framer-motion";

const App: React.FC = () => {
    
    const popover = usePopoverState();

    return (
        <div className="w-screen h-screen flex felx-col justify-center items-center p-4">
            <PopoverDisclosure
            as="div" 
            {...popover}>
                <h1 className="text-3xl text-gray-800 font-bold">
                    Abra Ibra
                </h1>
            </PopoverDisclosure>

            <AnimatePresence>
                {popover.visible &&
                <Popover
                aria-label="Abra"
                as={motion.div}
                initial="exited"
                animate="opened"
                exit="exited"
                variants={{
                    opened: {
                        opacity: 1
                    },
                    exited: {
                        opacity: 0
                    }
                }}
                {...popover} 
                visible>
                    Popover!!
                </Popover>}
            </AnimatePresence>
        </div>
    );
}

export default App;
```

**Reply by diegohaz** [reply]
I haven't tested that, but I imagine that will create some issues with focus management. The solution I posted on https://github.com/reakit/reakit/discussions/843#discussioncomment-467315 seems to better address that.

**Reply by joseDaKing** [reply]
Okay then I will go with your solution

**Comment by RedVelocity** [comment]
Hi,

How will this work with ComboboxPopover?
